backend/schema/errors.go

```diff
@@ -66,11 +67,12 @@ func ErrorListFromProto(e *schemapb.ErrorList) *ErrorList {
 	}
 }
 
-func Errorf(pos Position, endColumn int, format string, args ...any) Error {
-	return Error{Msg: fmt.Sprintf(format, args...), Pos: pos, EndColumn: endColumn}
+func Errorf(pos Position, endColumn int, format string, args ...any) *Error {
```

Comment:
Why change to a pointer?

Reply:
i structured the proto `ErrorList` to wrap a pointer list of errors `Errors []*Error` because this seemed most consistent with our other proto types. the updates to these other functions were just to make everything integrate with that in the way that felt smoothest, I think

is it preferable to use a value here? i can update if so